packages/block-library/src/comments-pagination-next/index.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Server-side rendering of the `core/comments-pagination-next` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/comments-pagination-next` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the next comments link for the query pagination.
+ */
+function render_block_core_comments_pagination_next( $attributes, $content, $block ) {
+	$comments_per_page = isset( $block->context['queryPerPage'] ) ? $block->context['queryPerPage'] : 4;
+	$comments_number   = (int) get_comments_number();
+	$max_page          = (int) floor( $comments_number / $comments_per_page );
+	$default_label     = __( 'Next Comments' );
+	$label             = isset( $attributes['label'] ) && ! empty( $attributes['label'] ) ? $attributes['label'] : $default_label;
+	$pagination_arrow  = get_comments_pagination_arrow( $block, true );
+	if ( $pagination_arrow ) {
+		$label .= $pagination_arrow;
+	}
+	$next_comments_link = get_next_comments_link( $label, $max_page );
```

Comment:
I couldn't make the pagination work as expected. I'm not sure the implementation here is correct as we have a custom `Comments query` and [get_next_comments_link](https://developer.wordpress.org/reference/functions/get_next_comments_link/) internally uses the global query `$page = get_query_var( 'cpage' );`

I guess we should take into account `get_approved_comments` that is used in the parent block to fetch the comments and we might probably need to make a new custom query ourselves, [similar to QueryPagination blocks](get_next_comments_link).

Reply:
Just noting that there were some recent changes in query pagination blocks that I'll have to check first, but the fact that I couldn't make this work as expected, still stands.

Reply:
I'm updating the code with a lot of changes that were requested if you prefer to wait for them. 

How is your discussion settings configured? Maybe we have a different configuration there.

Maybe we need the comment query loop working with pagination at first, although we have already the page numbers working, so I will keep taking a look at it to have it working as expected.

Regarding
> I guess we should take into account get_approved_comments that is used in the parent block to fetch the comments and we might probably need to make a new custom query ourselves, similar to QueryPagination blocks.

We started a discussion about that [right here](https://github.com/WordPress/gutenberg/issues/36642)! I think now is starting to get more relevant the requirement of having a multi-query loop for comments.

<img width="855" alt="Screenshot 2021-12-03 at 16 53 19" src="https://user-images.githubusercontent.com/37012961/144632351-acd5d918-063a-45f3-a634-095934f37fa8.png">

Reply:
Thanks for the review! It's helping quite a lot to move the comments pagination forward.

> I couldn't make the pagination work as expected. I'm not sure the implementation here is correct as we have a custom Comments query and get_next_comments_link internally uses the global query $page = get_query_var( 'cpage' );

You are adding the block outside a post or page? If this help you, I have these Discussion settings:

<img width="855" alt="Screenshot 2021-12-03 at 16 53 19" src="https://user-images.githubusercontent.com/37012961/144635046-33f9cacc-0b95-42f5-9a46-70c0b074fcb1.png">

If you are not and it is not working, we should also have to take a look at the Comment Pagination, as I think we are using there also the `cpage` query var.

> I guess we should take into account get_approved_comments that is used in the parent block to fetch the comments and we might probably need to make a new custom query ourselves, similar to QueryPagination blocks.

We started a discussion [about that here](https://github.com/WordPress/gutenberg/issues/36642)! Maybe this issue will help us understand the requirement of having multi-query comments like in the Query. I will add it to the conversation.